src/rinq/amqp/internal/amqputil/channel_pool.go

```diff
@@ -72,19 +120,90 @@ func (p *channelPool) GetQOS(preFetch uint) (*amqp.Channel, error) {
 }
 
 func (p *channelPool) Put(channel *amqp.Channel) {
+	p.put <- channel
```

Comment:
This will block forever if called after/during a shutdown.

Reply:
Good catch.